lib/autoupdate/agent/updater.go

```diff
@@ -279,7 +292,7 @@ func (u *Updater) Enable(ctx context.Context, override OverrideConfig) error {
 		if err != nil {
 			return trace.Errorf("failed to request version from proxy: %w", err)
 		}
-		desiredVersion = resp.AutoUpdate.AgentVersion
+		targetVersion = resp.AutoUpdate.AgentVersion
 		if resp.Edition == "ent" {
```

Comment:
not related to this change, as I understand we want isolate import of `modules` package where we have `BuildEnterprise` constant, may be we should add some log error if edition is unknown, might be helpful for debug if by any reason `BuildEnterprise` constant is modified